### PR TITLE
Respect leading whitespace in `readLiteral` and `matchLiteral`

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Per the discussion summarized here: https://github.com/chapel-lang/chapel/issues/19487#issuecomment-1591634119, update `readLiteral` and `matchLiteral` to respect leading whitespace in the literal string when `ignoreWhitespace=true`. 

- [x] local paratest
- [x] gasnet paratest

### CODE REVIEW

File: runtime/src/qio/qio_formatted.c [hunk 1]
```diff
@@ -1107,13 +1110,17 @@ qioerr qio_channel_scan_literal(const int threadsafe, qio_channel_t* restrict ch
     if( err ) return err;
   }
 
-  err = qio_channel_mark(false, ch);
+  err = qio_channel_mark(false, ch); //A
```
Comment (benharsh) [inline runtime/src/qio/qio_formatted.c:1113]: What are these 'A' and 'B' comments?

**Review by benharsh — APPROVED** [review]
This looks good to me, but can you also add some tests with ``ignoreLeadingWhitespace=false`` with literals and files that have different leading whitespace?

File: runtime/src/qio/qio_formatted.c (hunk 1 above)
Comment (jeremiah-corrado) [reply runtime/src/qio/qio_formatted.c:1113]: Just to keep track of which pairs of mark+revert/commit go together. I can remove them if it seems more confusing than helpful.

Comment (benharsh) [reply runtime/src/qio/qio_formatted.c:1113]: For me, the letters alone aren't immediately clear on an initial read-through, but I did assume it was some kind of pairing after reading some more. Maybe "begin mark A"? Anyways, I think it's fine either way.

**Pull request merged**